File: ams/switch_ams.go
package main

/*
	go tool compile -S switch_ams.go
	only for-each
*/
func IsSomething(num int) bool {
	switch num {
	case 1, 3, 5, 7, 9:
		return true
	default:
		return false
	}
}

func IsSomething2(num int) bool {
	switch num {
	case 1, 10, 100, 1000, 10000, 100000:
		return true
	default:
		return false
	}
}

func main() {
	IsSomething(10)
}

File: ams/switch_ams.c
#include <stdio.h>
// gcc -S -o target.s switch_ams.c
void is_something(int num) {
	
	switch(num) {

	case 1:
		break;
	case 2:
		break;
	case 3:
		break;
    case 4:
        break;
    case 5:
        break;
    case 6:
        break;
	default:
		break;
	}
}
 
int main () {
	
	is_something(10000);
   	return 0;
}
